src/components/analyticVisualization/EditVisualization.js

```diff
@@ -0,0 +1,96 @@
+import React, {useEffect, useState} from 'react'
+import i18n from '@dhis2/d2-i18n'
+import PropTypes from 'prop-types'
+import {
+    Button,
+    ButtonStrip,
+    Modal,
+    ModalActions,
+    ModalContent,
+    ModalTitle,
+} from '@dhis2/ui'
+import {NotEditableElement, VisualizationGroup, VisualizationTitle} from "./index";
+
+const EditVisualization = ({open, settings, handleChange, currentGroup, groups, handleEdit, handleClose, type}) => {
+    const [typeName, useTypeName] = useState('')
+    const [label, useLabel] = useState('')
+
+    const chooseTypeName = type => {
+        switch (type) {
+            case 'home':
+                useTypeName('')
+                useLabel('Home')
+                break
+            case 'program':
+                useTypeName('programName')
+                useLabel('Program')
+                break
+            case 'dataset':
+                useTypeName('datasetName')
+                useLabel('Data set')
+        }
+    }
+
+    useEffect(() => {
```

Comment:
It looks like type is just a prop to this component and not changeable from this component or its children, so it seems like the useEffect should not be necessary?